File: app/src/main/java/app/mumandroidassignment/lesson4/byjava/ShoppingActivity.java
package app.mumandroidassignment.lesson4.byjava;

import android.app.Activity;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.Toast;

import app.mumandroidassignment.R;

/**
 * Created by CodingHome on 4/5/18.
 */

public class ShoppingActivity extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_shopping);
    }

    public void onCategoryClick (View view){
        int id = view.getId();
        switch (id){
            case R.id.beauty:
                Toast.makeText(this,
                        "You have chosen the Beauty category of shopping",
                        Toast.LENGTH_SHORT).show();
                break;
            case R.id.clothing:
                Toast.makeText(this,
                        "You have chosen the Clothing category of shopping",
                        Toast.LENGTH_SHORT).show();
                break;
            case R.id.electronics:
                Toast.makeText(this,
                        "You have chosen the Electronics category of shopping",
                        Toast.LENGTH_SHORT).show();
                break;
            case R.id.food:
                Toast.makeText(this,
                        "You have chosen the Food category of shopping",
                        Toast.LENGTH_SHORT).show();
                break;
        }
    }
}

File: app/src/main/java/app/mumandroidassignment/lesson4/bykotlin/ui/ShoppingActivity.kt
package app.mumandroidassignment.lesson4.bykotlin.ui

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import app.mumandroidassignment.R
import app.mumandroidassignment.lesson5.bykotlin.question2.ui.activity.BeautyActivity
import app.mumandroidassignment.lesson5.bykotlin.question2.ui.activity.ClothingActivity
import app.mumandroidassignment.lesson5.bykotlin.question2.ui.activity.ElectronicActivity
import app.mumandroidassignment.lesson5.bykotlin.question2.ui.activity.FoodActivity
import kotlinx.android.synthetic.main.activity_shopping.*

class ShoppingActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_shopping)
        var username = intent.getStringExtra("username")
        tv_account.setText("welcome " + username)
    }

    fun onCategoryClick(view: View) {
        var id = view.id
        when (id) {
            R.id.beauty -> startActivity(Intent(this, BeautyActivity::class.java))
            R.id.clothing -> startActivity(Intent(this, ClothingActivity::class.java))
            R.id.electronics -> startActivity(Intent(this, ElectronicActivity::class.java))
            R.id.food -> startActivity(Intent(this, FoodActivity::class.java))
        }
    }

}
